src/usng.ts

```diff
@@ -0,0 +1,1396 @@
+//               2014 Mike Adair, Richard Greenwood, Didier Richard, Stephen Irons, Olivier Terral and Calvin Metcalf (proj4js)
+//               2014 Codice Foundation
+// Released under the MIT License; see
+// http://www.opensource.org/licenses/mit-license.php
+// or http://en.wikipedia.org/wiki/MIT_License
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+/*global define,Math,parseInt,parseFloat*/
+
+type LL = {
+  lat: number,
+  lon: number
+};
+
+type BBox = {
+  north: number,
+  south: number,
+  east: number,
+  west: number
+};
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/atanh
+const atanh = Math["atanh"] || function(x) {
+  return Math.log((1+x)/(1-x)) / 2;
+};
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/EPSILON
+const epsilon = typeof Number["EPSILON"] === "undefined" ? Math.pow(2, -52) : Number["EPSILON"];
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/hypot
+const hypot = Math["hypot"] || function (x, y) {
+  // https://bugzilla.mozilla.org/show_bug.cgi?id=896264#c28
+  var max = 0;
+  var s = 0;
+  for (var i = 0; i < arguments.length; i += 1) {
+    var arg = Math.abs(Number(arguments[i]));
+    if (arg > max) {
+      s *= (max / arg) * (max / arg);
+      max = arg;
+    }
+    s += arg === 0 && max === 0 ? 0 : (arg / max) * (arg / max);
+  }
+  return max === 1 / 0 ? 1 / 0 : max * Math.sqrt(s);
+};
+
+const includes = (list, item) => {
+  return list.indexOf(item) >= 0;
+};
+
+const sinh = Math["sinh"] || function(x) {
+  const y = Math.exp(x);
+  return (y - 1 / y) / 2;
+}
+
+/**
+ * This library was modified from the original usngs.js script to fix bugs, edge conditions, and limitations
+ * dealing with the different precision levels in usng/mgrs. It is by no means perfect! If you find a bug,
+ * submit it!
+ */
+function eatanhe(x) {
```

Comment:
:question: Can the following three functions be converted to `const` lambda(s) as well?

Reply:
Done.